### ISSUE TITLE
[SDK v1.4.1] generate_vnnx - ERROR: Node StatefulPartitionedCall/sequential_1/softmax/Softmax: input shape (1, 10) with axis -1 not supported for softmax

### ISSUE THREAD

**Issue by okawaj9** [Microchip-Vectorblox/VectorBlox-SDK#9]
Hi,

**Problem statement:**
- SDK v1.3.3 (w/ mo v2021.4.2) succeeded to execute generate_vnnx with attached "simple tensorflow2/keras model" (myfirstcnn saved_model) .
- But SDK v1.4.1 (w/ mo v2022.1.0) failed to execute  generate_vnnx with same model  (myfirstcnn saved_model) .

**Error Message (SDK v1.4.1)** :   ERROR: Node StatefulPartitionedCall/sequential_1/softmax/Softmax: input shape (1, 10) with axis -1 not supported for softmax

**Issue Replication (operation):**
mo --saved_model_dir  ./myfirstcnn \
        --input_shape [1,28,28,1] \
        --scale_values [255.0] \
        --static_shape

generate_vnnx -x saved_model.xml  -c V1000 -f ../../sample_images -o myfirstcnn.vnnx

**Attachment:**
[myfirstcnn.zip](https://github.com/Microchip-Vectorblox/VectorBlox-SDK/files/9703068/myfirstcnn.zip)

Could you please let me know the workaround with SDK v1.4.1 ?

Best Regards,

**Comment by cirqueit** [comment]
Our code currently does not parse negative indexing properly for Softmax. Please try the following fix:

```diff
--- a/sdk/python/vbx/vbx/generate/onnx_convert.py
+++ b/sdk/python/vbx/vbx/generate/onnx_convert.py
@@ -1679,7 +1679,7 @@ def gen_softmax(vinode):

     batch_size_axis = 0
     for i,s in enumerate(idims):
-        if i==axis or i==batch_size_axis:
+        if i==(axis % len(idims)) or i==batch_size_axis:
             continue
         if s == 1:
             continue
```

**Comment by okawaj9** [comment]
Many thanks.

- By the patch, the "generate_vnnx" issue was resolved.

- On the other hand, simulation with "$VBX_SDK/example/python/classifier.py" on SDK 1.4.1 caused following error which I saw with v1.3.1.  Is it possible to take a look into it, please ?


python $VBX_SDK/example/python/classifier.py  myfirstcn.vnnx   ../../test_images/seven.28.jpg

Traceback (most recent call last):
  File "/home/okawa/Microchip/VBX/SDK/VectorBlox-SDK-release-v1.4.1/example/python/classifier.py", line 50, in <module>
    output = vnnx_infer(args.model, input_array)[0]
  File "/home/okawa/Microchip/VBX/SDK/VectorBlox-SDK-release-v1.4.1/example/python/classifier.py", line 25, in vnnx_infer
    outputs = model.run([flattened])
  File "/home/okawa/Microchip/VBX/SDK/VectorBlox-SDK-release-v1.4.1/python/vbx/vbx/sim/model.py", line 85, in run
    c_shim.vbx_cnn_model_start(Model.vbx_cnn,self.model_bytes,io_buffers)
  File "/home/okawa/Microchip/VBX/SDK/VectorBlox-SDK-release-v1.4.1/python/vbx/vbx/sim/c.py", line 66, in vbx_cnn_model_start
    raise TypeError("Input {} expected {} got {}".
TypeError: Input 0 expected int8 got uint8

Best Regards,

**Comment by okawaj9** [comment]
Attached another simple network which is using CNN and Dense.  It worked with v1.3.3, but failed in v1.4.1 with the "TypeError: Input 0 expected int8 got uint8" error.

[myfirstcnn3.zip](https://github.com/Microchip-Vectorblox/VectorBlox-SDK/files/9721461/myfirstcnn3.zip)

Best Regards,

**Comment by okawaj9** [comment]
Confirmed this issue was resolved with v1.4.2.  Then let me close this.
Many thanks !
